kod.go

```diff
@@ -220,6 +208,11 @@ func (wc *WithGlobalConfig[T]) getGlobalConfig() any {
 	return &wc.config
 }
 
+// Core is initialized by the Kod instance.
+type Core interface {
+	Init(context.Context, *Kod) error
+}
```

Comment:
Unless I'm not wrong. There is nothing currently implementing.

I'm a bit lost with the purpose of the Init method. Not by its functional need,but my its name and parameter

You pass kod as argument, but then Init is a strange name: ConfigureKod, ApplyKod, BootKod, InjectKod 

To me, Init name is about initialization of the struct.

Here you are injecting Kod, right? But a pointer, so it could apply modifications to Kod

Here I would have seen something inverted

func(context.Context) (func(*Kod) error)

I'm unsure, but I'm raising the point

Reply:
I'm considering the extend for kod to support lifecycle management for other dependencies like slog/otel or else, with Init and Shutdown function.

And this pr is just WIP status right now :-)